kanvas/src/jsMain/kotlin/HtmlCanvas.kt

```diff
@@ -0,0 +1,226 @@
+) : Kanvas<Paint, Path2D> {
+
+    private val context = element.getContext("2d") as CanvasRenderingContext2D
+
+    override val width: Float
+        get() = context.canvas.width.toFloat()
+
+    override val height: Float
+        get() = context.canvas.height.toFloat()
+
+    override fun buildPaint(paint: Paint): Paint = paint
+
+    override fun buildPath(actions: PathBuilder<*>.() -> Unit): Path2D =
+        Path2DBuilder().apply(actions).build()
+
+    /** TODO: This can be cleaned up. */
+    private fun applyPaint(paint: Paint): Unit = when (paint) {
+        is Paint.Stroke -> {
+            context.strokeStyle = "rgba(${paint.color.red}, ${paint.color.green}, ${paint.color.blue}, ${paint.color.alpha})"
+            context.lineWidth = paint.width.toDouble()
+            context.lineCap = when (paint.cap) {
+                Paint.Stroke.Cap.Butt -> CanvasLineCap.BUTT
+                Paint.Stroke.Cap.Round -> CanvasLineCap.ROUND
+                Paint.Stroke.Cap.Square -> CanvasLineCap.SQUARE
+            }
+            context.lineJoin = when (paint.join) {
+                Paint.Stroke.Join.Bevel -> CanvasLineJoin.BEVEL
+                Paint.Stroke.Join.Round -> CanvasLineJoin.ROUND
+                is Paint.Stroke.Join.Miter -> {
+                    context.miterLimit = paint.join.limit.toDouble()
+                    CanvasLineJoin.MITER
+                }
+            }
+        }
+        is Paint.Fill -> {
+            context.fillStyle = "rgba(${paint.color.red}, ${paint.color.green}, ${paint.color.blue}, ${paint.color.alpha})"
+        }
+        is Paint.Text -> {
+            context.fillStyle = "rgba(${paint.color.red}, ${paint.color.green}, ${paint.color.blue}, ${paint.color.alpha})"
+            context.textAlign = when (paint.alignment) {
+                Paint.Text.Alignment.Left -> CanvasTextAlign.LEFT
+                Paint.Text.Alignment.Center -> CanvasTextAlign.CENTER
+                Paint.Text.Alignment.Right -> CanvasTextAlign.RIGHT
+            }
+            context.font = "${paint.size}px ${paint.font.names.joinToString { "\"$it\"" }}"
+        }
+    }
+
+    override fun drawArc(
+        left: Float,
+        top: Float,
+        right: Float,
+        bottom: Float,
+        startAngle: Float,
+        sweepAngle: Float,
+        paint: Paint,
+    ) {
+        require(paint !is Paint.Text)
+        applyPaint(paint)
+        context.beginPath()
+        val radiusX = (right - left) / 2.0
+        val radiusY = (bottom - top) / 2.0
+        val x = left + radiusX
+        val y = top + radiusY
+        val startAngleRadians = startAngle * PI / 180f
+        val endAngleRadians = (startAngle + sweepAngle) * PI / 180f
+        context.ellipse(x, y, radiusX, radiusY, rotation = 0.0, startAngleRadians, endAngleRadians)
+        when (paint) {
+            is Paint.Stroke -> context.stroke()
+            is Paint.Fill -> context.fill()
+            else -> error("unreachable")
```

Comment:
Is it possible to suppress rather than having this deadcode `else`?
In either case, might be nice to have an comment explaining that it is unreachable because of `require` at beginning of function covering the last `Paint` type?